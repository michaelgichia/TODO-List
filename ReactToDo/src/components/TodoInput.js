import React, { Component, PropTypes } from 'react';

export default class TodoInput extends Component {
	render() {
		return(
			<div>
				<form onSubmit={this.props.onSubmit} className="z-depth-2">
					<input
						type="text"
						value={this.props.value}
						onChange={this.props.onChange}
						ref={this.props.refs}
						placeholder="Enter to do..."/>
				</form>
			</div>
		)
	}
}

TodoInput.propTypes = {
	onSubmit: PropTypes.func.isRequired,
	onChange: PropTypes.func.isRequired,
	value: PropTypes.any,
	refs: PropTypes.any
}
