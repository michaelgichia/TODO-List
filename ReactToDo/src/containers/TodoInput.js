import React, { Component } from 'react';
import _ from 'lodash';

export default class TodoInput extends Component {
	constructor(props) {
		super(props)
		this.state = {
			term: ''
		}
	}
  onInputChange = (event) => {
    this.setState({term: event.target.value})
  }

  handelSubmit = (event) => {
    event.preventDefault();
    if(this.refs.inputValue.value.length < 5) {return}
    const value = this.refs.inputValue.value
    this.props.createTodo(value)
    this.setState({term: ""})
  }

	render() {
		return(
			<div>
				<form onSubmit={this.handelSubmit} className="z-depth-2">
					<input
						type="text"
						value={this.state.term}
						onChange={this.onInputChange}
						ref="inputValue"
						placeholder="Enter to do..."/>
				</form>
			</div>
		)
	}
}

