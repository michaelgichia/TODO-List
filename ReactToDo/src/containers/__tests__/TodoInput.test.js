import React from 'react';
import { mount, shallow } from 'enzyme';

import TodoInput from '../TodoInput';

describe('TodoInput components', () => {
  const props = {};
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<TodoInput {...props}/>);
  });

  it('renders correctly', () => {
    const tree = shallow(<TodoInput />);
    expect(tree).toMatchSnapshot();
  });
});