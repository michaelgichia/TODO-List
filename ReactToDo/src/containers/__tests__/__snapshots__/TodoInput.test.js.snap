// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInput components renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <form
        className="z-depth-2"
        onSubmit={[Function]}
    >
        <input
            onChange={[Function]}
            placeholder="Enter to do..."
            type="text"
            value=""
        />
    </form>
</div>,
  "nodes": Array [
    <div>
      <form
            className="z-depth-2"
            onSubmit={[Function]}
      >
            <input
                  onChange={[Function]}
                  placeholder="Enter to do..."
                  type="text"
                  value=""
            />
      </form>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <TodoInput />,
      "_debugID": 6,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": TodoInput {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "handelSubmit": [Function],
        "onInputChange": [Function],
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "term": "",
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 4,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <form
                    className="z-depth-2"
                    onSubmit={[Function]}
          >
                    <input
                              onChange={[Function]}
                              placeholder="Enter to do..."
                              type="text"
                              value=""
                    />
          </form>
</div>,
        "_debugID": 7,
        "_renderedOutput": <div>
          <form
                    className="z-depth-2"
                    onSubmit={[Function]}
          >
                    <input
                              onChange={[Function]}
                              placeholder="Enter to do..."
                              type="text"
                              value=""
                    />
          </form>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TodoInput />,
}
`;
